### ISSUE TITLE
Guardhouse and smithy not getting added to street upon creation

### ISSUE THREAD

**Issue by ryceg** [ryceg/Eigengrau-s-Essential-Establishment-Generator#630]
In the Road profile, creating a guardhouse will not make the table update.

**Comment by rurikid** [comment]
Under the same conditions this also seems to be affecting the smithy. I tested all other options and these seem to be the only two affected.

**Comment by Saxeen** [comment]
I've been looking into this and I believe I managed to resolve the issue for the smithy by making a few modifications to the smithy building creation script based on ones from buildings that worked as intended. I must admit I don't fully understand all the code I modified, but it looks like the reason for the table not updating is that the building creation code didn't take in the road data as an optional parameter. Looking at the whole town buildings list after generating a building, smithys/guardhouses do get created, they're just not connected to the right road but instead randomizes a new one.
I've however been unable to reproduce the same type of solution for the guardhouse. The building creation code is different enough that I can't just directly reuse the same lines.
smithy changes for reference: https://github.com/Saxeen/Eigengrau-s-Essential-Establishment-Generator/commit/16b32a3b5b8899f02d03bc3e83ea4d6e5b17c105
